common/domain-setter.sub.html

```diff
@@ -0,0 +1,8 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>A page that will likely be same-origin-domain but not same-origin</title>
+
+<script>
+"use strict";
+document.domain = "{{host}}";
```

Comment:
I think it's the same when running under the correct configuration, e.g. when navigating to `http://web-platform.test:8000`. But when running at `http://localhost:8000`, they will be different. So `"{{host}}"` seems more correct.

I don't understand the scenario you're describing. If I grab the prototype object before setting document.domain, it will be null, and null doesn't change to non-null.